extensions/spark/src/main/scala/ai/djl/spark/task/BasePredictor.scala

```diff
@@ -78,9 +80,13 @@ abstract class BasePredictor[A, B](override val uid: String) extends Transformer
    */
   def setTranslator(value: Translator[A, B]): this.type = set(translator, value)
 
+  def setTranslatorFactory(value: TranslatorFactory): this.type = set(translatorFactory, value)
+
+  def setTokenizer(value: String): this.type = set(tokenizer, value)
+
   /** @inheritdoc */
   override def transform(dataset: Dataset[_]): DataFrame = {
-    model = new ModelLoader[A, B]($(engine), $(modelUrl), $(inputClass), $(outputClass))
+    model = new ModelLoader[A, B]($(engine), $(modelUrl), $(inputClass), $(outputClass), $(translator), $(translatorFactory), $(tokenizer))
```

Comment:
1. I think Tokenizer should be a member of transaltorFactory.
2. Let's only expose translatorFactory for now, it's a bit confusing have both translator and translatorFactory

Reply:
Changed.